cmake/Modules/FindThrift.cmake

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#[=======================================================================[.rst:
+FindThrift
+-------
+
+Finds the Apache Thrift library.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+This module provides the following imported targets, if found:
+
+``Thrift::thrift``
+  The Apache Thrift library
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``Thrift_FOUND``
+  True if the system has the Thrift library.
+``Thrift_VERSION``
+  The version of the Thrift library which was found.
+``Thrift_INCLUDE_DIRS``
+  Include directories needed to use Thrift.
+``Thrift_LIBRARIES``
+  Libraries needed to link to Thrift.
+
+Cache Variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``Thrift_INCLUDE_DIR``
+  The directory containing ``thrift/Thrift.h``.
+``Thrift_LIBRARY``
+  The path to the Thrift library.
+
+#]=======================================================================]
+
+find_path(
+    Thrift_INCLUDE_DIR
+    NAMES thrift/Thrift.h
+    PATH_SUFFIXES include
+)
+
+find_library(
```

Comment:
should you put this after `find_package_handle_standard_args `?

Reply:
I followed the tutorial at https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html.

`find_package_handle_standard_args` does the necessary checks at the end, and prints logs depending on whether the libraries and include directories were found successfully.